Level 2

ArtemPraktikum opened the pull request:
gl hf

ArtemPraktikum committed:
feat: initial commit + сссылка на фронт в readmi

ArtemPraktikum committed:
feat: основная структура проекта

ArtemPraktikum committed:
feat: комп. app, шрифты, нормалайз

ArtemPraktikum committed:
feat: сделать фулл адаптивный хедер с сайд меню

ArtemPraktikum committed:
feat: Promo + navtab с якорями

ArtemPraktikum committed:
feat: aboutproject адаптив

ArtemPraktikum committed:
feat: techs адаптив

ArtemPraktikum committed:
feat: AboutMe, Portfolio адаптив

ArtemPraktikum committed:
feat: footer адаптив

ArtemPraktikum committed:
feat: половина movies и весь searchform адаптив

ArtemPraktikum committed:
feat:  вся страница /movies и компоненты:
movies, searchform, moviescardlist, moviescard, prelouder. адаптив всё.

ArtemPraktikum committed:
feat: saved-movies (это копия movies с др. роутом)

ArtemPraktikum committed:
feat: страница\компонент profile адаптив

ArtemPraktikum committed:
feat: стр/компонент register адаптив

ArtemPraktikum committed:
feat: стр/комп. login адаптив

ArtemPraktikum committed:
feat: комп/стр. NotFoundPage адаптив

ArtemPraktikum committed:
fix: удалить border

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
`КНопка формы поиска фильмов обрезается по краям`
<img width="185" alt="Снимок экрана 2022-07-21 в 11 32 32" src="https://user-images.githubusercontent.com/25247802/180169104-aff154cd-81c9-4c46-9d35-3248f96b0808.png">

`Отсутствует состояние кнопки в карточке "крестик»`


- [ ] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
`Появляется горизонтальный скролл на расширении от 600рх и менее`
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вклxaадке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [ ] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`Данный пункт возможно корректно проверить только после устранения горизонтального скролла`
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
`Весь контент между header и footer нужно заключить в тег main, как в тег для уникального контента страницы.`
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
	- [x] <!--3.38--> разметка закчлючена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
`Кнопка формы поиска фильмов`
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы формы поиска фильмов не выделяются, когда на них установлен ФОКУС.`
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [ ] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

### Рекомендации
`Неправильно задан lang`

## Количество баллов: 60

ArtemPraktikum committed:
fix: исправить сплющивание кнопки поиска в поиске
и обернуть глав. стр. в тег main

ArtemPraktikum committed:
big fix: изменить подход к адаптивности всего:
а именно отказаться от резиновости в угоду pixel perfect версте
без гориз. прокрутке на всем сайте. на всех разрешениях.
и фикс login.css и login.js

ArtemPraktikum committed:
fix: исправить язык сайта в index.html

ArtemPraktikum commented:
насколько я понял пункт: " Для каждого шрифта указаны альтернативные варианты из системных шрифтов." изначально был выполнен т.к. в файле app.css есть подключение альтернативных шрифтов для всей страницы

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
`Появляется горизонтальный скролл на расширении от 600рх и менее`
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вклxaадке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [ ] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`Кнопка формы поиска фильмов имеет большой отступ от формы`
<img width="229" alt="Снимок экрана 2022-07-22 в 11 03 08" src="https://user-images.githubusercontent.com/25247802/180394451-b6bcd6d8-c9c7-4735-84cf-ba838cec94c0.png">

`Отсутствует отзывчивая верстка на промежуточных расширениях. Не должно быть больших отступов по краям.`
<img width="752" alt="Снимок экрана 2022-07-22 в 11 03 29" src="https://user-images.githubusercontent.com/25247802/180394544-175c1a8a-0b9d-4161-8a12-f90c22ae7a94.png">

- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
`Весь контент между header и footer нужно заключить в тег main, как в тег для уникального контента страницы.`
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
	- [x] <!--3.38--> разметка закчлючена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
`Кнопка формы поиска фильмов`
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы формы поиска фильмов не выделяются, когда на них установлен ФОКУС.`
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [ ] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;



## Количество баллов: 85
